WooCommerce/Classes/ViewRelated/Products/ProductSearchViewControllerStateCoordinator.swift

```diff
@@ -0,0 +1,53 @@
+import UIKit
+
+/// UI state of `ProductSearchViewController`.
+///
+/// - noResultsPlaceholder: a no results placeholder is displayed
+/// - syncing: syncing data UI
+/// - results: the results are shown
+enum ProductSearchViewControllerState {
```

Comment:
`ProductSearchViewControllerStateCoordinator` seems long. What do you think of `ProductSearchStateCoordinator` ?